docs/runtime_image.md

```diff
@@ -0,0 +1,83 @@
+
+To make this work S2I needs to know the following:
+
+* builder image
+* mapping of the artifacts
+* runtime image
+
+And all these information you should specify during S2I invocation:
+
+    s2i build <repo> <builder-image> <my-app> --runtime-image <runtime-image> --runtime-artifact </path/to/artifact>
+
+The only new here are `--runtime-image` and `--runtime-artifact` options. The latter option is used to specify a full path to file (or directory) inside of the builder image and a corresponding directory on the runtime image where S2I will copy these files. Note that you can specify this option multiple times.
+
+For our example S2I will do the following steps:
+
+1. runs builder container and performs a build (as usual)
+1. after build finishes but before stopping runtime container, it downloads artifacts from the builder and places them to a temporary directory on the host
+1. runs container with runtime image
+1. uploads scripts and build artifacts from a temporary directory into `WORKDIR` of a runtime container
+1. runs `assemble-runtime` script in a runtime container
+1. commits container with runtime image
+1. stops build and runtime containers
+
+## Details
+
+### Format and behavior of `--runtime-artifact` option
+
+`--runtime-artifact` option have a format `<source>[:<destination>]`. Here is an example of sample values:
+
+| Value                   | Meaning |
+|-------------------------|---------|
+| `/tmp/app.war`          | `/tmp/app.war` file will be extracted from a builder image and uploaded into the `WORKDIR` of a runtime image |
+| `/tmp/app.war:.`        | the same as above |
+| `/tmp/app.war:app/dist` | `/tmp/app.war` file will be copied from a builder image into into the `app/dist` subdirectory of the `WORKDIR` of a runtime image |
+| `/opt/data`             | `/opt/data` directory will be copied from a builder image into the `WORKDIR` of a runtime image |
+
+Keep in mind that the `source` must be an absolute path. The `destination` must be a relative path to directory, it must exist and path must not start from `..` 
+
+During copying the artifacts, S2I will modify their permissions. All directories and files with executable bit will be uploaded with `0755` mode. Other files will have `0644` mode.
+
+### `assemble-runtime` script requirements
+
+`assemble-runtime` script can be written in any programming language. S2I searches the following locations for this script in the following order:
+
+1. A script found at the `--scripts-url` URL
+1. A script found in the application source `.s2i/bin` directory
+1. A script found at the default image URL (`io.openshift.s2i.scripts-url` label)
+
+`assemble-runtime` script is always gets executed under `USER` from a runtime image.
+
+### Runtime image requirements
+
+In most of the cases you can use any images as a runtime image. Only if you are using `--allowed-uids` option then image should have numeric `USER` and this user should be within the range of allowed users.
```

Comment:
"In most cases you can use any image as a runtime image.  However, if you are using the `--allowed-uids` option then the image must have a numeric `USER` specified and the value must be within the range of allowed user ids."